File: app/src/main/java/com/alvarlagerlof/koda/Comments/CommentsSend.java
package com.alvarlagerlof.koda.Comments;

import android.content.Context;
import android.os.AsyncTask;
import android.util.Log;
import android.widget.Toast;

import com.alvarlagerlof.koda.Cookies.PersistentCookieStore;
import com.alvarlagerlof.koda.RemoteConfigValues;

import java.io.IOException;
import java.net.CookieHandler;
import java.net.CookieManager;
import java.net.CookiePolicy;

import okhttp3.FormBody;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.internal.JavaNetCookieJar;

/**
 * Created by alvar on 2017-01-28.
 */

class CommentsSend extends AsyncTask<Void, Void, String> {

    private Context context;
    private String url;
    private String comment;

    CommentsSend(Context context, String publicID, String comment) {
        this.context = context;
        this.url = RemoteConfigValues.URL_COMMENTS_SEND + publicID;
        this.comment = comment;
    }


    @Override
    protected String doInBackground(Void... params) {

        Log.d("URL", url);


        CookieHandler cookieHandler = new CookieManager(
                new PersistentCookieStore(context), CookiePolicy.ACCEPT_ALL);

        OkHttpClient client = new OkHttpClient.Builder()
                .cookieJar(new JavaNetCookieJar(cookieHandler))
                .build();

        FormBody formBody = new FormBody.Builder()
                .add("comment", comment)
                .build();

        Request request = new Request.Builder()
                .url(url)
                .post(formBody)
                .build();


        try {
            Response response = client.newCall(request).execute();
            Log.d("RESPONSE", response.body().string());
            Toast.makeText(context, response.body().string(), Toast.LENGTH_LONG).show();


            response.body().close();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return null;
    }
}

File: app/src/main/java/com/alvarlagerlof/koda/Comments/CommentsSend.kt
package com.alvarlagerlof.koda.Comments

import android.content.Context
import android.os.AsyncTask
import android.util.Log
import android.widget.Toast
import com.alvarlagerlof.koda.Cookies.PersistentCookieStore
import com.alvarlagerlof.koda.Vars
import okhttp3.FormBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.internal.JavaNetCookieJar
import java.io.IOException
import java.net.CookieManager
import java.net.CookiePolicy

/**
 * Created by alvar on 2017-01-28.
 */

internal class CommentsSend(private val context: Context, publicID: String, private val comment: String) : AsyncTask<Void, Void, String>() {
    private val url: String = Vars.URL_COMMENTS_SEND + publicID


    override fun doInBackground(vararg params: Void): String? {

        Log.d("URL", url)


        val cookieHandler = CookieManager(
                PersistentCookieStore(context), CookiePolicy.ACCEPT_ALL)

        val client = OkHttpClient.Builder()
                .cookieJar(JavaNetCookieJar(cookieHandler))
                .build()

        val formBody = FormBody.Builder()
                .add("comment", comment)
                .build()

        val request = Request.Builder()
                .url(url)
                .post(formBody)
                .build()


        try {
            val response = client.newCall(request).execute()
            Log.d("RESPONSE", response.body()!!.string())
            Toast.makeText(context, response.body()!!.string(), Toast.LENGTH_LONG).show()


            response.body()!!.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return null
    }
}

